lease acquisition uses non-idempotent txn closure, probably breaking OSC

apply this diff:

```diff
diff --git a/server/testserver.go b/server/testserver.go
index 2fad316..5e7f5ea 100644
--- a/server/testserver.go
+++ b/server/testserver.go
@@ -18,6 +18,8 @@
 package server

 import (
+       "time"
+
        "github.com/cockroachdb/cockroach/client"
        "github.com/cockroachdb/cockroach/config"
        "github.com/cockroachdb/cockroach/gossip"
@@ -62,7 +64,7 @@ func NewTestContext() *Context {
        // MaxOffset is the maximum offset for clocks in the cluster.
        // This is mostly irrelevant except when testing reads within
        // uncertainty intervals.
-       ctx.MaxOffset = 0
+       ctx.MaxOffset = 50 * time.Millisecond

        // Load test certs. In addition, the tests requiring certs
        // need to call security.SetReadFileFn(securitytest.Asset)
```

It leads to some transaction restarts (which would also happen in production).
Then,

```
--- FAIL: TestLeaseManager (0.03s)
        lease_test.go:90: expected /1/1, but found
--- FAIL: TestLeaseManagerReacquire (0.02s)
        lease_test.go:253: expected refcount of 2, but found 3
```

The issue is that `(*LeaseManager).Acquire` takes a transaction, but changes its internal state according to operations carried out within that transaction without ascertaining that the transaction actually commits (which, of course, it may not).

The resulting state looks quite inconsistent, with refcount overcounting being on the harmless side. In the first test failure, it appears that the node thinks it has a lease while nothing's been persisted to the lease table, breaking online schema changes.

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -2822,7 +2822,6 @@ func TestChangeReplicasDuplicateError(t *testing.T) {
 // we don't erroneously return that descriptor (recently fixed bug) if the
 func TestRangeDanglingMetaIntent(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	t.Skip("TODO(tschottdorf): https://github.com/cockroachdb/cockroach/issues/3020")
 	// Test RangeLookup with Scan.
 	testRangeDanglingMetaIntent(t, false)
 	// Test RangeLookup with ReverseScan.
@@ -2864,16 +2863,27 @@ func testRangeDanglingMetaIntent(t *testing.T, isReverse bool) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	pArgs := putArgs(keys.RangeMetaKey(roachpb.RKey(key)), data)
 	txn := newTransaction("test", key, 1, roachpb.SERIALIZABLE, tc.clock)
+	// Officially begin the transaction. If not for this, the intent resolution
+	// machinery would simply remove the intent we write below, see #3020.
+	// We send directly to Replica throughout this test, so there's no danger
+	// of the Store aborting this transaction (i.e. we don't have to set a high
+	// priority).
+	bArgs, h := beginTxnArgs(key, txn)
+	if _, err := client.SendWrappedWith(tc.Sender(), tc.rng.context(), h, &bArgs); err != nil {
+		t.Fatal(err)
+	}
 
+	pArgs := putArgs(keys.RangeMetaKey(roachpb.RKey(key)), data)
+	txn.Sequence++
 	if _, err := client.SendWrappedWith(tc.Sender(), tc.rng.context(), roachpb.Header{Txn: txn}, &pArgs); err != nil {
 		t.Fatal(err)
 	}
 
 	// Now lookup the range; should get the value. Since the lookup is
-	// inconsistent, there's no WriteIntentErorr.
-	rlArgs.Key = keys.RangeMetaKey(roachpb.RKey("A"))
+	// inconsistent, there's no WriteIntentError.
+	// Note that 'A' < 'a'.
+	rlArgs.Key = keys.RangeMetaKey(roachpb.RKey{'A'})
 
 	reply, err = client.SendWrappedWith(tc.Sender(), tc.rng.context(), roachpb.Header{
 		Timestamp:       roachpb.MinTimestamp,

